### ISSUE TITLE
Dev branch list index out of range

### ISSUE THREAD

**Issue by hellcatwatermelonfriedchicken** [tayler6000/pyVoIP#175]
Had to manually put the files into my python3 packages folder, got dev branch working and changed to using credential manager etc.

Getting this error now:

(Only using dev branch hoping the 30 second call end issue fixes, not sure if it will)
```py
Exception in thread VoIPSocket Thread:
Traceback (most recent call last):
  File "/usr/lib/python3.11/threading.py", line 1038, in _bootstrap_inner
    self.run()
  File "/usr/local/lib/python3.11/dist-packages/pyVoIP/sock/sock.py", line 333, in run
    message = SIPMessage(data)
              ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/dist-packages/pyVoIP/SIP/message.py", line 316, in __init__
    self.parse(data)
  File "/usr/local/lib/python3.11/dist-packages/pyVoIP/SIP/message.py", line 352, in parse
    self.parse_sip_request(data)
  File "/usr/local/lib/python3.11/dist-packages/pyVoIP/SIP/message.py", line 783, in parse_sip_request
    self.version = self.heading[2]
                   ~~~~~~~~~~~~^^^
IndexError: list index out of range
```

**Comment by hellcatwatermelonfriedchicken** [comment]
Getting this as well, it's kind of swapping between the two errors:
Added a print for the args to bind, seems to not be an argument error but according to google the invalid argument error is generic and used for multiple things

('0.0.0.0', 5060)
Exception in thread VoIPSocket Thread:
Traceback (most recent call last):
  File "/usr/lib/python3.11/threading.py", line 1038, in _bootstrap_inner
    self.run()
  File "/usr/local/lib/python3.11/dist-packages/pyVoIP/sock/sock.py", line 325, in run
    self.bind((self.bind_ip, self.bind_port))
  File "/usr/local/lib/python3.11/dist-packages/pyVoIP/sock/sock.py", line 316, in bind
    self.s.bind(addr)
OSError: [Errno 22] Invalid argument

**Comment by tayler6000** [comment]
@hellcatwatermelonfriedchicken I just noticed the `IndexError: list index out of range` in the pytest logs in the GitHub workflow, but I don't seem to get that on my computer.  Could you run a packet capture and send the SIP message that's causing this error?

**Comment by hellcatwatermelonfriedchicken** [comment]
Wasn't able to reproduce this anymore, but getting this issue now:
CSeq: {'check': '102', 'method': 'BYE'}
User-Agent: voicecall
Proxy-Authorization: Digest username="rah", realm="x.voip.ms", algorithm=MD5, uri="sip:x.voip.ms", nonce="086055f5", response="fa02cb023c08d6aa4f101b7179e4ae0e"
X-Asterisk-HangupCause: No user responding
X-Asterisk-HangupCauseCode: 18
Content-Length: 0

Body:

Raw:
b'BYE sip:rah@0.0.0.0:5060 SIP/2.0\r\nVia: SIP/2.0/UDP 208.100.60.8:5060;branch=z9hG4bK39610d4c;rport\r\nMax-Forwards: 70\r\nFrom: <sip:x.voip.ms>;tag=as311a06ae\r\nTo: <sip:x@0.0.0.0>;tag=db29078c\r\nCall-ID: 4b227777d4dd1fc61c6f884f48641d02@0.0.0.0:5060\r\nCSeq: 102 BYE\r\nUser-Agent: voicecall\r\nProxy-Authorization: Digest username="281267_rah", realm="chicago.voip.ms", algorithm=MD5, uri="sip:chicago1.voip.ms", nonce="086055f5", response="fa02cb023c08d6aa4f101b7179e4ae0e"\r\nX-Asterisk-HangupCause: No user responding\r\nX-Asterisk-HangupCauseCode: 18\r\nContent-Length: 0\r\n\r\n'
BYE recieved

Using dev branch

**Comment by tayler6000** [comment]
> Getting this as well, it's kind of swapping between the two errors: Added a print for the args to bind, seems to not be an argument error but according to google the invalid argument error is generic and used for multiple things
> 
> ```python
> ('0.0.0.0', 5060)
> Exception in thread VoIPSocket Thread:
> Traceback (most recent call last):
>   File "/usr/lib/python3.11/threading.py", line 1038, in _bootstrap_inner
>     self.run()
>   File "/usr/local/lib/python3.11/dist-packages/pyVoIP/sock/sock.py", line 325, in run
>     self.bind((self.bind_ip, self.bind_port))
>   File "/usr/local/lib/python3.11/dist-packages/pyVoIP/sock/sock.py", line 316, in bind
>     self.s.bind(addr)
> OSError: [Errno 22] Invalid argument
> ```

This was fixed in #220

**Comment by tayler6000** [comment]
Closing as original issue is resolved and new issue is a duplicate of #226
